### ISSUE TITLE
Completions not refreshed?

### ISSUE THREAD

**Issue by lnicola** [zimfw/zimfw#243]
```
Zim commit ref:   c501a1b
Zsh version:      zsh 5.4.2 (x86_64-unknown-linux-gnu)
System info:      Linux ubik.home 4.14.12-1-ARCH #1 SMP PREEMPT Fri Jan 5 18:19:34 UTC 2018 x86_64 GNU/Linux
```

- gist of .zimrc: https://github.com/lnicola/dotfiles/blob/master/.zimrc
- gist of .zshrc: https://github.com/lnicola/dotfiles/blob/master/.zshrc

Description
-----------
I realized today that some completions don't work. Running `compinit` fixed it, but I think the `completion` module should be doing that automatically.

```text
$ ll .zcompdump*
-rw-r--r-- 1 grayshade grayshade  47K Feb  3  2016 .zcompdump
-r--r--r-- 1 grayshade grayshade 101K Feb  3  2016 .zcompdump.zwc
$ compinit
$ ll .zcompdump*
-rw-r--r-- 1 grayshade grayshade  42K Jan 11 22:08 .zcompdump
-r--r--r-- 1 grayshade grayshade 101K Feb  3  2016 .zcompdump.zwc
```

Am I missing something?

**Comment by lnicola** [comment]
https://github.com/zimfw/zimfw/blob/master/modules/completion/init.zsh#L19

Does `-C` mean that it should skip rebuilding the file? Am I supposed to manually run that?

**Comment by Plloi** [comment]
if your failing completions are related to the archive or autosuggestions modules, it could be because completions should be last in the zmodules array

**Comment by lnicola** [comment]
No, it's about files in `/usr/share/zsh/site-functions`. I believe `compinit -C` doesn't re-read those.

**Comment by Plloi** [comment]
Could you post your .zlogin?

**Comment by Plloi** [comment]
Relevant blurb from zsh docs

> Should you need to use the original completion commands, you can still bind keys to the old widgets by putting a '.' in front of the widget name, e.g. '.expand-or-complete'.
> To speed up the running of compinit, it can be made to produce a dumped configuration that will be 
 read in on future invocations; this is the default, but can be turned off by calling compinit with the option -D. The dumped file is .zcompdump in the same directory as the startup files (i.e. $ZDOTDIR or $HOME); alternatively, an explicit file name can be given by 'compinit -d dumpfile'. The next invocation of compinit will read the dumped file instead of performing a full initialization.
> 
> If the number of completion files changes, compinit will recognise this and produce a new dump file. However, if the name of a function or the arguments in the first line of a #compdef function (as described below) change, it is easiest to delete the dump file by hand so that compinit will re-create it the next time it is run. The check performed to see if there are new functions can be omitted by giving the option -C. In this case the dump file will only be created if there isn't one already.

**Comment by lnicola** [comment]
On one box I don't have it, on another one it's:

```zsh
#
# startup file read in interactive login shells
#
# The following code helps us by optimizing the existing framework.
# This includes zcompile, zcompdump, etc.
#

(
  # Function to determine the need of a zcompile. If the .zwc file
  # does not exist, or the base file is newer, we need to compile.
  # These jobs are asynchronous, and will not impact the interactive shell
  zcompare() {
    if [[ -s ${1} && ( ! -s ${1}.zwc || ${1} -nt ${1}.zwc) ]]; then
      zcompile ${1}
    fi
  }

  zim_mods=${ZDOTDIR:-${HOME}}/.zim/modules
  setopt EXTENDED_GLOB

  # zcompile the completion cache; siginificant speedup.
  for file in ${ZDOTDIR:-${HOME}}/.zcomp^(*.zwc)(.); do
    zcompare ${file}
  done

  # zcompile .zshrc
  zcompare ${ZDOTDIR:-${HOME}}/.zshrc

  # zcompile some light module init scripts
  zcompare ${zim_mods}/git/init.zsh
  zcompare ${zim_mods}/utility/init.zsh
  zcompare ${zim_mods}/pacman/init.zsh
  zcompare ${zim_mods}/spectrum/init.zsh
  zcompare ${zim_mods}/completion/init.zsh
  zcompare ${zim_mods}/fasd/init.zsh

  # zcompile all .zsh files in the custom module
  for file in ${zim_mods}/custom/**/^(README.md|*.zwc)(.); do
    zcompare ${file}
  done

  # zcompile all autoloaded functions
  for file in ${zim_mods}/**/functions/^(*.zwc)(.); do
    zcompare ${file}
  done

  # syntax-highlighting
  for file in ${zim_mods}/syntax-highlighting/external/highlighters/**/*.zsh; do
    zcompare ${file}
  done
  zcompare ${zim_mods}/syntax-highlighting/external/zsh-syntax-highlighting.zsh

  # zsh-histery-substring-search
  zcompare ${zim_mods}/history-substring-search/external/zsh-history-substring-search.zsh
```

**Comment by PatTheMav** [comment]
@lnicola That looks like the old `.zlogin` - it was changed a while back to use the new `zrecompile` functionality. Update that file with the more recent version (https://github.com/zimfw/zimfw/blob/master/templates/zlogin), then maybe run `zmanage clean-cache && zmanage build-cache`, finally reload/restart your terminal session.

(This new `zlogan` file will then also be kept up2date as it's sourced from `zimfw`'s directory).

The guys are working on optimizing this behavior, but using `zmanage` to update your completion caches is a acceptable trade-off for `zimfw`'s speed benefits.

**Comment by lnicola** [comment]
On the third box I had the new version of `.zlogin`.

> The guys are working on optimizing this behavior, but using zmanage to update your completion caches is a acceptable trade-off for zimfw's speed benefits.

All right, then.

**Comment by PatTheMav** [comment]
@lnicola But did it fix your issue then?
